### PR TITLE
Feat: Average Blending

### LINKED ISSUE #112: Add more blending types
- Add
- Multiple

What others are standard types of blending?

### LINKED ISSUE #115: Add average blending
Blend fragment =  sum of fragments colors/ number of samples.
We might have to clamp the final color in case gets > 1.0

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
Adds logic for blending the average of each model.

The GLSL `mix()` function mixes based on a percentage (0 - 1) of the two inputs. The ratio of the first input (`v_sample.a`) to the total (`v_sample.a + m_sample.a`) yields that percentage. 

- closes #112 
- closes #115

### CODE REVIEW

**Review by broarr — APPROVED** [review]

File: src/aframe/fragment-shader.frag [hunk 1]
```diff
@@ -136,34 +138,35 @@ void main() {
...
-                        model_sample = sample_model(model_structs[i], data_position);
-                        volume_sample = mix(volume_sample, model_sample, mix_factor);
+                        m_sample = sample_model(model_structs[i], data_position);
+                        v_sample = mix(v_sample, m_sample, mix_factor);
                     }
                 }
                 #pragma unroll_loop_end
             }
-        } else break; // array is "empty", leave vFragColor transparent
+        } else break; // No models in use, leave transparent
```
Comment (kmilo9999) [inline src/aframe/fragment-shader.frag:166]: place  `model_sample.rgb *= model.intensity;`  here. So the value of the pixel only gets increased once per model and not once per sample.

File: src/aframe/fragment-shader.frag [hunk 2]
```diff
@@ -136,34 +138,35 @@ void main() {
 
     // Starting from the entry point, march the ray through the volume and sample it
     vec4 vFragColor = vec4(0);
-    vec4 model_sample, volume_sample;
+    vec4 m_sample, v_sample;
```
Comment (kmilo9999) [inline src/aframe/fragment-shader.frag:141]: `m_sample ` is only used inside the internal for loop. you can move it below `if(blending != 0)`

**Review by kmilo9999 — COMMENTED** [review]
👍  keep the good work !

**Review by kmilo9999 — CHANGES_REQUESTED** [review]
Adding comments to my review

File: example/src/App.js [hunk 3]
```diff
@@ -98,7 +98,7 @@ function App() {
   const VV = (
     <StyledVolumeViewer
       controlsVisible={controlsVisible}
-      blending={Blending.Max}
```
[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply example/src/App.js:101]: Will do!

File: src/aframe/fragment-shader.frag (hunk 1 above)
Comment (RobertGemmaJr) [reply src/aframe/fragment-shader.frag:169]: Actually yeah I refactored a little more to calculate the mix factor correctly and it doesn't actually make a difference? Should each model have its own intensity or should it be a property of the volume itself that's calculated after all of the blending is done?

File: src/aframe/fragment-shader.frag (hunk 2 above)
Comment (RobertGemmaJr) [reply src/aframe/fragment-shader.frag:141]: Moved it

File: src/aframe/fragment-shader.frag (hunk 1 above)
Comment (kmilo9999) [reply src/aframe/fragment-shader.frag:173]: it should be calculated after all the blending is done. let's apply the max intensity of the models.

Comment (RobertGemmaJr) [reply src/aframe/fragment-shader.frag:175]: Okay it should be all set!

**Comment by RobertGemmaJr** [comment]
@eldu can you take a look at the shader code while Camilo is on vacation? Thank you!

**Comment by RobertGemmaJr** [comment]
Consolidating with other Blending changes to #131

**Review by eldu — COMMENTED** [review]
Added in some comments and questions! Looks good to me though !

**Pull request closed without merging**